Include mu-plugins in site exports

bgrgicak opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "Fixes" keyword and use "Related to" instead.
-->

## Proposed Changes

- Include `mu-plugins` in Studio exports.
- Exclude the SQLite plugin located in mu-plugins from Studio exports.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

- Checkout this branch
- Create a site
- Add a mu-plugin to your site (it can be an empty PHP file)
- Export site
- Confirm the mu-plugin is included in the site export
- Confirm there is no SQLite plugin in mu-plugins
- (Optional) Import site into Studio
- Confirm the site works

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Have you checked for TypeScript, React or other console errors?

bgrgicak committed:
Include mu-plugins in site exports

bgrgicak committed:
Fix unit tests

bgrgicak committed:
Use muPlugins as the category name

bgrgicak committed:
Exclude paths in getBackupContents

fredrikekelund commented on src/lib/import-export/export/exporters/default-exporter.ts:
```diff
@@ -219,22 +221,32 @@ export class DefaultExporter extends EventEmitter implements Exporter {
 		};
 
 		const siteFiles = await this.getSiteFiles();
+		const pathsToExclude = [ 'wp-content/mu-plugins/sqlite-database-integration' ];
```
Can we move this to the top level? Like a `readonly` prop on the class

fredrikekelund commented on src/lib/import-export/export/exporters/default-exporter.ts:
```diff
@@ -219,22 +221,32 @@ export class DefaultExporter extends EventEmitter implements Exporter {
-			const [ wpContent, category ] = relativePathItems;
+			const [ wpContent, wpContentDirectory ] = relativePathItems;
 
+			if ( pathsToExclude.some( ( path ) => relativePath.startsWith( path ) ) ) {
+				return;
+			}
```
I feel like the `getSiteFiles` method might be a better fit for this logic. Thoughts?

fredrikekelund reviewed (changes requested):
The code and functionality here LGTM overall 👍 However, there's an issue where sites that still have the old mu-plugins (the ones we moved to Playground's internal folder in https://github.com/Automattic/studio/pull/845) will now fail to export if they aren't started before exporting.

Definitely an edge case, but I imagine it'd be relatively simple to address.

bgrgicak committed:
Exclude unwanted paths in getSiteFiles

bgrgicak committed:
Exclude outdated internal mu-plugins

bgrgicak commented:
> However, there's an issue where sites that still have the old mu-plugins (the ones we moved to Playground's internal folder in https://github.com/Automattic/studio/pull/845) will now fail to export if they aren't started before exporting.

Good catch @fredrikekelund! I added these files to the list and also addressed your other comments. 
The outdated internal plugins should exist only until WP-CLI is executed for the first time or the site is started, but there is still a chance that the user will click export before the WP-CLI command runs (there is a command that runs every time a site screen is loaded in Studio to check the version).

fredrikekelund committed:
Windows support

fredrikekelund committed:
Test one more excluded plugin

fredrikekelund reviewed (approved):
I took the liberty of pushing a change that fixes an issue with the excluded mu-plugins on Windows. Aside from that, this LGTM 👍 Feel free to tweak my changes if needed, @bgrgicak.

> The outdated internal plugins should exist only until WP-CLI is executed for the first time or the site is started, but there is still a chance that the user will click export before the WP-CLI command runs

Yeah, this is definitely an edge case, but it _can_ happen, so I think implementing this kind of protection makes sense.

fredrikekelund commented:
@bgrgicak, please note that there's currently a code freeze until v1.3.3 is released (see pfHvTO-lI-p2), so please hold off with merging this for now.
